src/features/TodoList/TodoListItem.jsx

```diff
@@ -0,0 +1,59 @@
+import { useState } from 'react';
+import TextInputWithLabel from '../../shared/TextInputWithLabel';
+
+function TodoListItem({ todo, onCompleteTodo, onUpdateTodo }) {
+  const [isEditing, setIsEditing] = useState(false);
+
+  const [workingTitle, setWorkingTitle] = useState(todo.title);
+
+  function handleCancel() {
+    setWorkingTitle(todo.title);
+    setIsEditing(!isEditing);
```

Comment:
This is clever! But you **have** to be sure that this will not trigger any bugs. Since, this functionality of this is to set a state value to `false` and not necessarily to create a toggle affect, it's best to leave it as `setState(false)`.